### PR TITLE
run tests with python 3.12

### PR DESCRIPTION
**Pull request by geier**

### CODE REVIEW

**Comment by geier** [comment]
blocked by https://github.com/aio-libs/aiohttp/issues/7646

File: tox.ini [hunk 1]
```diff
@@ -15,15 +15,10 @@ passenv =
...
-deps =
-    pytest
-    freezegun
-    !py36: vdirsyncer
-    py36: vdirsyncer==0.16.8
-    python-dateutil
+extras =
+    tests
     pytz20187: pytz==2018.7
     pytz_latest: pytz
```
Comment (WhyNotHugo) [inline tox.ini:21]: These should remain in `deps =`:

```suggestion
deps =
    pytz20187: pytz==2018.7
    pytz_latest: pytz
```

File: tox.ini [hunk 2]
```diff
@@ -15,15 +15,10 @@ passenv =
   TRAVIS_JOB_NUMBER
   TRAVIS_PULL_REQUEST
   TRAVIS_REPO_SLUG
-deps =
-    pytest
```
Comment (WhyNotHugo) [inline tox.ini:19]: `pytest` is missing from `project.optional-dependencies.test`. This is why CI is failing.

File: tox.ini [hunk 3]
```diff
@@ -15,15 +15,10 @@ passenv =
...
-    freezegun
-    !py36: vdirsyncer
-    py36: vdirsyncer==0.16.8
-    python-dateutil
```
Comment (WhyNotHugo) [inline tox.ini:23]: This one is also missing, but is not required because it is a non-test dependency already.

**Review by WhyNotHugo — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 3 above)
Comment (WhyNotHugo) [reply tox.ini:23]: (nothing to do on this item, just FYI)

File: tox.ini (hunk 2 above)
Comment (geier) [reply tox.ini:19]: Thanks. It looks like we still need to define all dependencies in tox.ini or make it run pip install independently. Not sure, which way is worse. For now I optioned for using commands_pre to run pip install .[test].

Comment (WhyNotHugo) [reply tox.ini:19]: Here you're overriding dependencies, so they need to be in the `deps =` array (e.g.: these two are not extras).

Fixing this tests work.

**Pull request merged**